mllib/src/test/scala/org/apache/spark/mllib/clustering/LDASuite.scala

```diff
@@ -99,9 +99,15 @@ class LDASuite extends SparkFunSuite with MLlibTestSparkContext {
 
     // Check: per-doc topic distributions
     val topicDistributions = model.topicDistributions.collect()
+
     //  Ensure all documents are covered.
-    assert(topicDistributions.length === tinyCorpus.length)
-    assert(tinyCorpus.map(_._1).toSet === topicDistributions.map(_._1).toSet)
+    // SPARK-5562. since the topicDistribution returns the distribution of docsI over topics .
+    // for empty docs, since the distribution of topic won't sum to 1 (and hence it is not a pdf)
+    // So the output will not contain the empty docs and hence we modify the unittest to
+    // compare against nonEmptyTinyCorpus
+    val nonEmptyTinyCorpus = getNonEmptyDoc(tinyCorpus)
+    assert(topicDistributions.length === nonEmptyTinyCorpus.length)
```

Comment:
I disagree; there should be topicDistributions for each empty document, probably set to the prior for the corpus.  I think this points to a deeper problem.  I'll comment on that elsewhere.